### PR TITLE
Make API swagger doc accessible only from back office

### LINKED ISSUE #36929: Make the admin API documentation private
# Context
- Unauthenticated users must not be able to access to the API documentation from the URL `admin-api/docs.html`
- The URL `admin-api/docs.html` is no longer accessible

# Tests
GIVEN a user (authenticated or not on the BO)
WHEN: he brows the following URL : `/admin-api/docs.html`
THEN: the documentation is not accessible (a 404 error occured instead)

GIVEN: an unauthenticated user on the back office
WHEN: he brows the following URL : `/admin-dev/configure/advanced/admin-api/docs.html`
THEN: he is redirected to the login page

GIVEN: an authenticated user on the back office
WHEN: he brows the following URL : `admin-dev/configure/advanced/admin-api/docs.html`
THEN: the documentation is accessible

### Technical details
- In the API kernel the swagger UI must be disabled https://github.com/PrestaShop/PrestaShop/blob/3b10746eceaa1c9633850d5ddb84e245c7685905/app/config/admin-api/config.yml#L17
- In the Admin kernel the swagger UI must be enabled in https://github.com/PrestaShop/PrestaShop/blob/a464b5c22c2e342d8bb415c1108cf1776447aaf4/app/config/admin/config.yml#L16-L15 (maybe there is a need to integrate some routing file from the APIPlatform bundle)
- The two links in the `Admin API` BO section must be updated to open the new URLs

### PR DESCRIPTION
**Pull request by jolelievre**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Make API swagger doc accessible only from back office
| Type?             | improvement
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | See issue
| UI Tests          | https://github.com/jolelievre/ga.tests.ui.pr/actions/runs/10943804293
| Fixed issue or discussion?     | Fixes #36929
| Related PRs       | ~
| Sponsor company   | ~

### CODE REVIEW

**Review by Progi1984 — APPROVED** [review]
Ok for UI Tests

**Review by nicosomb — APPROVED** [review]

**Review by matthieu-rolland — APPROVED** [review]

**Review by florine2623 — APPROVED** [review]
Hello @jolelievre ,

**Before :**

With a user (authenticated or not on the BO)
`/admin-api/docs.html`
-> accessible

With a user (authenticated or not on the BO)
`/admin-dev/configure/advanced/admin-api/docs.html`
-> Exception found
Screenshot 2024-09-23 at 10 37 44

**After :**

Following the tests cases below.

_GIVEN_ a user (authenticated or not on the BO)
_WHEN_: he brows the following URL : `/admin-api/docs.html`
_THEN_: the documentation is not accessible (a 404 error occurred instead) ✅ 

Screenshot 2024-09-23 at 10 54 55
Screenshot 2024-09-23 at 10 53 58

_GIVEN_: an unauthenticated user on the back office
_WHEN_: he brows the following URL : `/admin-dev/configure/advanced/admin-api/docs.html`
_THEN_: he is redirected to the login page ✅ 

_GIVEN_: an authenticated user on the back office
_WHEN_: he brows the following URL : `/admin-dev/configure/advanced/admin-api/docs.html`
_THEN_: the documentation is accessible ✅ 

Working as expected.
Thanks!

**Comment by ps-jarvis** [comment]
PR merged, well done!

Message to @PrestaShop/committers: do not forget to milestone it before the merge.

**Pull request merged**
